dev/com.ibm.ws.messaging.comms.client/src/com/ibm/ws/sib/jfapchannel/impl/CommsOutboundChain.java

```diff
@@ -127,89 +159,130 @@ void unbindSecureFacet(OutboundSecureFacet oldFacet) {
         }
     }
 
-    private Map<String, Object> getTcpOptions() {
+    public Map<String, Object> getTcpOptions() {
         if (isAnyTracingEnabled() && tc.isEntryEnabled()) Tr.entry(this, tc, "getTcpOptions");
         Map<String, Object> tcpProps = tcpOptions.getConfiguration();
         if (isAnyTracingEnabled() && tc.isEntryEnabled()) exit(this, tc, "getTcpOptions", tcpProps);
         return tcpProps;
     }
 
+    @Reference(name = "nettyTlsProvider", cardinality = OPTIONAL, policyOption = GREEDY, unbind = "unbindTlsProviderService")
+    void bindNettyTlsProviderService(NettyTlsProvider tlsProvider) {
+        if (isAnyTracingEnabled() && tc.isEntryEnabled()) Tr.entry(this, tc, "bindTlsProviderService", tlsProvider);
+        this.nettyTlsProvider = tlsProvider;
+        if (isAnyTracingEnabled() && tc.isEntryEnabled()) exit(this, tc, "bindTlsProviderService");
+    }
+
+    void unbindTlsProviderService(NettyTlsProvider oldService) {
+        if (isAnyTracingEnabled() && tc.isEntryEnabled()) entry(this, tc, "unbindTlsProviderService", oldService);
+        // TODO: Figure out if there's something to be done here
+        if (isAnyTracingEnabled() && tc.isEntryEnabled()) exit(this, tc, "unbindTlsProviderService");
+    }
+
+    public NettyTlsProvider getNettyTlsProvider() {
+        if (isAnyTracingEnabled() && tc.isEntryEnabled()) Tr.entry(this, tc, "getNettyTlsProvider");
+        if (isAnyTracingEnabled() && tc.isEntryEnabled()) exit(this, tc, "getNettyTlsProvider", nettyTlsProvider);
+        return nettyTlsProvider;
+    }
+
     private synchronized void createBasicJFAPChain() {
         if (isAnyTracingEnabled() && tc.isEntryEnabled()) entry(this, tc, "createBasicJFAPChain", chainName);
-        try {
-            ChannelFramework cfw = commsClientService.getChannelFramework();
-            cfw.registerFactory("JFapChannelOutbound", JFapChannelFactory.class);
+        if(!useNetty()) {
```

Comment:
Avoid using `if (!condition) {} else {}` — flip the condition and swap the if/else blocks. 
This should be a refactoring in your IDE.